messaging/src/main/java/org/axonframework/common/caching/EhCacheAdapter.java

```diff
@@ -63,11 +65,31 @@ public boolean remove(Object key) {
         return ehCache.remove(key);
     }
 
+    @Override
+    public void removeAll() {
+        ehCache.removeAll();
+    }
+
     @Override
     public boolean containsKey(Object key) {
         return ehCache.isKeyInCache(key);
     }
 
+    @Override
+    public synchronized <V> void computeIfPresent(Object key, UnaryOperator<V> update) {
```

Comment:
ehCache does have some atomic operations to replace an item with another one. Instead of synchronizing access to this method, we could use `ehCache.replace()` to replace the current element with the updated one atomically.

Reply:
Although true, I don't necessarily see how that allows me to perform an update on the entry we replace.
Just for clarity, that API looks like this:

- `replace(new Element)`
- `replace(old Element, new Element)`

Or, am I missing something obvious / awesome that we can do with this?

Reply:
Discussed offline, *and* had some time to think. We can make a workable solution with `replace` to get this solved.
Only outlier I'm not overly happy with, is when the `update` task changes the old value to `null`. 
EhCache will still insert the entry in that case, which is different from regular computeIfPresent operations.

I've pushed a change to make it work, though; let me know what you think.